### PR TITLE
Integration tests

### PR DESCRIPTION
**Pull request by flavio**
more tests

### CODE REVIEW

File: spec/patches_operations_spec.rb [hunk 1]
```diff
@@ -189,6 +189,63 @@
       expect(exception.stdout).to include('security patches')
       expect(exception.stdout).to include('patches needed')
     end
+  end
+
+  context "analyze a running container" do
+    before :all do
+      @keep_alpine = docker_image_exists?("alpine", "latest")
+      if !@keep_alpine
```
Comment (mssola) [inline spec/patches_operations_spec.rb]:
```ruby
pull_image("alpine:latest") unless @keep_alpine
```

File: spec/patches_operations_spec.rb [hunk 2]
```diff
@@ -189,6 +189,63 @@
...
+      start_background_container(Settings::VULNERABLE_IMAGE, @vul_container)
+      @containers_to_terminate << @vul_container
+
+      expect(docker_image_exists?(@patched_image_repo, @patched_image_tag)).to be true
+      start_background_container(@patched_image, @patched_container)
+      @containers_to_terminate << @patched_container
+
+      start_background_container("alpine:latest", @not_suse_container)
+      @containers_to_terminate << @not_suse_container
+
```
Comment (mssola) [inline spec/patches_operations_spec.rb]: remove empty line

File: spec/update_operations_spec.rb [hunk 3]
```diff
@@ -16,37 +16,99 @@
...
+          Settings::VULNERABLE_IMAGE,
+        @patched_image)
+      }.to raise_error(Cheetah::ExecutionFailed)
+    end
   end
 
+  context "analyze a running container" do
+    before :all do
+      @keep_alpine = docker_image_exists?("alpine", "latest")
+      if !@keep_alpine
```
Comment (mssola) [inline spec/update_operations_spec.rb]: same as before

File: spec/update_operations_spec.rb [hunk 4]
```diff
@@ -16,37 +16,99 @@
...
+      start_background_container(Settings::VULNERABLE_IMAGE, @vul_container)
+      @containers_to_terminate << @vul_container
+
+      expect(docker_image_exists?(@patched_image_repo, @patched_image_tag)).to be true
+      start_background_container(@patched_image, @patched_container)
+      @containers_to_terminate << @patched_container
+
+      start_background_container("alpine:latest", @not_suse_container)
+      @containers_to_terminate << @not_suse_container
+
```
Comment (mssola) [inline spec/update_operations_spec.rb]: same as before

**Comment by mssola** [comment]
LGTM. Just fix my nitpicks ;)

**Comment by flavio** [comment]
I fixed the issues you reported, plus I added tests for `zypper-docker images`

**Pull request merged**